src/components/vuestic-components/va-infinite-scroll/VaInfiniteScroll.vue

```diff
@@ -0,0 +1,226 @@
+<template>
+  <component
+    :is="tag"
+    class="va-infinite-scroll"
+    :class="{'va-infinite-scroll--reversed': reverse}"
+  >
+    <slot
+      name="default"
+      ref="content"
+    />
+    <slot
+      name="loading"
+      v-if="!disabled"
+    >
+      <div ref="defaultSpinner"
+           class="va-infinite-scroll__spinner"
+           :class="{'va-infinite-scroll__spinner--invisible': !fetching}">
+        <va-progress-circle
+          size="small"
+          :thickness="0.15"
+          :color="error ? $themes.danger : $themes.primary"
+          indeterminate
+        />
+      </div>
+    </slot>
+  </component>
+</template>
+
+<script>
+import * as _ from 'lodash'
+import VaProgressCircle from '../va-progress-bar/progress-types/VaProgressCircle'
+import { makeContextablePropsMixin } from '../../context-test/context-provide/ContextPlugin'
+import { sleep } from '../../../services/utils'
+
+export default {
+  name: 'VaInfiniteScroll',
+  components: { VaProgressCircle },
+  mixins: [
+    makeContextablePropsMixin({
+      offset: {
```

Comment:
Any reason to have default offset at 1? Does it matter if we use 0? Also I see quasar is using 500 as default. Why won't we?

Might also make sense to double-check with refs if loader size is included in offset or not and act accordingly.